R/compare.R

```diff
@@ -101,12 +146,54 @@ compare <- function(x, y, ...,
     ignore_attr = ignore_attr,
     ignore_encoding = ignore_encoding,
     ignore_formula_env = ignore_formula_env,
-    ignore_function_env = ignore_function_env
+    ignore_function_env = ignore_function_env,
+    ignore_name_order = ignore_name_order,
+    ignore_NULLs = ignore_NULLs,
+    ignore_private = ignore_private
   )
+
+  opts$user_specified <- c(intersect(names(match.call()),
+                                   names(opts)),
+                           "user_specified")
+
   out <- compare_structure(x, y, paths = c(x_arg, y_arg), opts = opts)
   new_compare(out, max_diffs)
 }
 
+sort_named_parts <- function(x) {
```

Comment:
Could you please move these functions underneath `compare_structure()`?  We generally use the opposite of C order, where the big picture functions come first.

Reply:
Done